controllers/common_test.go

```diff
@@ -133,19 +133,64 @@ var _ = Describe("GetGeneratedJobName", func() {
 		BeforeEach(func() {
 			// UID without hyphens is 32.
 			maxNameLen = 30
+			optional = "optional.string"
+		})
+
+		It("Equals the truncated required string", func() {
+			Expect(generatedResourceName).To(Equal(required[:maxNameLen]))
+		})
+
+		It("Length does not exceed maxNameLen", func() {
+			Expect(len(generatedResourceName)).To(BeNumerically("<=", maxNameLen))
+		})
+
+		It("Is contained in the full hyphen-less UID", func() {
+			Expect(required).To(ContainSubstring(generatedResourceName))
+		})
+	})
+})
+
+var _ = Describe("GetGeneratedJobName", func() {
+
+	var (
+		uid            types.UID
+		objectMetaName string
+		maxNameLen     int
+
+		uidWithoutHyphens string
+
+		generatedJobName      string
+		generatedResourceName string
+	)
+
+	BeforeEach(func() {
+		uid = types.UID(uuid.New().String())
+	})
+
+	JustBeforeEach(func() {
+		uidWithoutHyphens = strings.Replace(string(uid), "-", "", -1)
+		generatedJobName = GetGeneratedJobName(uid, objectMetaName, maxNameLen)
+		generatedResourceName = GetGeneratedJobName(uid, objectMetaName, maxNameLen)
```

Comment:
These variables are assigned the same value? I think you mean to call `GetGeneratedResourceName` with `uidWithoutHyphens`? This makes the test on 192 a very happy test that won't ever fail :)

If you are only using `generatedResourceName` in one test, it might make more sense to create it in that test alone.

Reply:
oh yes, good catch. Fixed. 

It is a little tricky to move the creation of `generatedResourceName` since the `uid` gets generated a bunch of times for each spec. 
I tried to create the `generatedResourceName` within the `Before Each` block of the test itself (inside the `When`). 
This causes the `generatedJobName` and `GetGeneratedResourceName` to be called with different `uid` and `uidWithoutHyphens`. An example of the failure log below: 

```
•••••••••••••generatedJobName is object.meta.name-bced7caf409947faa7d2161963a1bff7
generatedResourceName is : object.meta.name-
•generatedJobName is object.meta.name-3c0b1cadf1e6465d9043f791d56f8d46
generatedResourceName is : object.meta.name-bced7caf409947faa7d2161963a1bff7
•generatedJobName is object.meta.name-2e70d1f94c12493f8797310821130c58
generatedResourceName is : object.meta.name-3c0b1cadf1e6465d9043f791d56f8d46
---------------------
    Expected
        <string>: object.meta.name-2e70d1f94c12493f8797310821130c58
    to be equivalent to
        <string>: object.meta.name-3c0b1cadf1e6465d9043f791d56f8d46
```

Reply:
Hmm.. I don't know how this issue was caused. It seems that the two `BeforeEach`s are called in the wrong order, which means that the UID from the last test and this test are used.

I wouldn't put it in a `BeforeEach` (which runs before a set of tests), and simply place it in the single test where it is used, i.e.:

```go
It("returns the same name as GetGeneratedResourceName with uidWithoutHyphens as the required string", func() {
        generatedResourceName := GetGeneratedResourceName(uidWithoutHyphens, objectMetaName, maxNameLen)
	Expect(generatedJobName).To(BeEquivalentTo(generatedResourceName))
})
```

This is pretty minor, so fine leaving as is.